### PR TITLE
HPCC-16185 Allow Dali file access auth when no scope scanning enabled

### PR DESCRIPTION
**Pull request by RussWhitehead**
Currently, in order for users to be authenticated when they access a file
in Dali, an LDAP Server has to be bound and scope scanning enabled. This PR
enables authentication even when scope scanning is not enabled

Signed-off-by: Russ Whitehead william.whitehead@lexisnexis.com

### CODE REVIEW

File: dali/server/daldap.cpp [hunk 1]
```diff
@@ -121,44 +121,46 @@ class CDaliLdapConnection: public CInterface, implements IDaliLdapConnection
...
-                        perm=ldapsecurity->authorizeWorkunitScope(*user, obj);
-                    if (perm==-1)
-                        perm = 0;
-                }
-            }
+            if (filescope)
+                perm=ldapsecurity->authorizeFileScope(*user, obj);
+            else if (wuscope)
+                perm=ldapsecurity->authorizeWorkunitScope(*user, obj);
+            if (perm==-1)
```
Comment (rpastrana) [inline dali/server/daldap.cpp]: Can we use meaningful value enumerations?

File: dali/server/daldap.cpp [hunk 2]
```diff
@@ -121,44 +121,46 @@ class CDaliLdapConnection: public CInterface, implements IDaliLdapConnection
...
+        }
+        else
+        {
+            WARNLOG("NULL UserDescriptor in daldap.cpp getPermissions('%s')",key ? key : "NULL");
+        }
+
+        if (0 == username.length())
+        {
+            username.append(filesdefaultuser);
+            decrypt(password, filesdefaultpassword);
```
Comment (rpastrana) [inline dali/server/daldap.cpp]: is filesdefaultpassword always != null?

**Comment by rpastrana** [comment]
@RussWhitehead just a couple of minor comments.

File: dali/server/daldap.cpp (hunk 1 above)
Comment (RussWhitehead) [inline dali/server/daldap.cpp]: These have already been updated to use the Sec_Access* enumerations in a later build (this is 5.6.8)

File: dali/server/daldap.cpp (hunk 2 above)
Comment (RussWhitehead) [inline dali/server/daldap.cpp]: default user/pw is optional so this can be NULL. In this case, authentication will fail and the user will be denied access

**Comment by RussWhitehead** [comment]
@richardkchapman Ready for merge into 5.6.8

File: dali/server/daldap.cpp (hunk 1 above)
Comment (richardkchapman) [inline dali/server/daldap.cpp]: That means we are going to get an upmerge clash...

Can you create a separate PR for 6.0.6 please

**Pull request merged**